### PR TITLE
Elasticache Module Version Upgrade

### PR DESCRIPTION
**Pull request by pk0331**
Elasticache Module Version Upgrade

### CODE REVIEW

File: modules/aws/elasticacheRedis/main.tf [hunk 1]
```diff
@@ -16,8 +17,8 @@ module "redis" {
   at_rest_encryption_enabled       = var.at_rest_encryption_enabled
   transit_encryption_enabled       = var.transit_encryption_enabled
   cloudwatch_metric_alarms_enabled = var.cloudwatch_metric_alarms_enabled
-
-  security_group_rules = [
+  allow_all_egress = false
+  additional_security_group_rules = [
```
Comment (suryaoruganti) [inline modules/aws/elasticacheRedis/main.tf:21]: why this change?

**Comment by shubhamgrg04** [comment]
@suryaoruganti tested it by creating an instance and connecting in preprod, can be merged
 
Screenshot 2022-12-22 at 4 46 20 PM

**Comment by suryaoruganti** [comment]
Is this backward compatible? If an instance is created with an older
version and the new version is applied, what is the tf plan output?

**Comment by shubhamgrg04** [comment]
> 

Yes, verified that it is backward compatiable by upgrading an instance in preprod. Following is the tf plan:

```
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  + create
  ~ update in-place
  - destroy
Terraform will perform the following actions:
  # module.redis.aws_cloudwatch_metric_alarm.cache_cpu[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "cache_cpu" {
        id                        = "ecredisvalidate10-001-cpu-utilization"
      ~ tags                      = {
          + "Name" = "ecredisvalidate10"
        }
      ~ tags_all                  = {
          + "Name"                  = "ecredisvalidate10"
            # (4 unchanged elements hidden)
        }
        # (17 unchanged attributes hidden)
    }
  # module.redis.aws_cloudwatch_metric_alarm.cache_memory[0] will be updated in-place
  ~ resource "aws_cloudwatch_metric_alarm" "cache_memory" {
        id                        = "ecredisvalidate10-001-freeable-memory"
      ~ tags                      = {
          + "Name" = "ecredisvalidate10"
        }
      ~ tags_all                  = {
          + "Name"                  = "ecredisvalidate10"
            # (4 unchanged elements hidden)
        }
        # (17 unchanged attributes hidden)
    }
  # module.redis.aws_elasticache_parameter_group.default[0] will be updated in-place
  ~ resource "aws_elasticache_parameter_group" "default" {
        id          = "ecredisvalidate10"
        name        = "ecredisvalidate10"
      ~ tags        = {
          + "Name" = "ecredisvalidate10"
        }
      ~ tags_all    = {
          + "Name"                  = "ecredisvalidate10"
            # (4 unchanged elements hidden)
        }
        # (3 unchanged attributes hidden)
    }
  # module.redis.aws_elasticache_replication_group.default[0] will be updated in-place
  ~ resource "aws_elasticache_replication_group" "default" {
        id                            = "ecredisvalidate10"
      ~ security_group_ids            = [
          - "sg-0c34d941a010d7315",
        ] -> (known after apply)
        tags                          = {
            "Name" = "ecredisvalidate10"
        }
        # (32 unchanged attributes hidden)
        # (1 unchanged block hidden)
    }
  # module.redis.aws_elasticache_subnet_group.default[0] will be updated in-place
  ~ resource "aws_elasticache_subnet_group" "default" {
      ~ description = "Managed by Terraform" -> "Elasticache subnet group for ecredisvalidate10"
        id          = "ecredisvalidate10"
        name        = "ecredisvalidate10"
      ~ tags        = {
          + "Name" = "ecredisvalidate10"
        }
      ~ tags_all    = {
          + "Name"                  = "ecredisvalidate10"
            # (4 unchanged elements hidden)
        }
        # (2 unchanged attributes hidden)
    }
  # module.redis.module.aws_security_group.aws_security_group.cbd[0] will be created
  + resource "aws_security_group" "cbd" {
      + arn                    = (known after apply)
      + description            = "Security group for Elasticache Redis"
      + egress                 = (known after apply)
      + id                     = (known after apply)
      + ingress                = (known after apply)
      + name                   = (known after apply)
      + name_prefix            = "ecredisvalidate10-"
      + owner_id               = (known after apply)
      + revoke_rules_on_delete = false
      + tags                   = {
          + "Name" = "ecredisvalidate10"
        }
      + tags_all               = {
          + "Name"                  = "ecredisvalidate10"
          + "argonaut.dev/env/dev3" = "true"
          + "argonaut.dev/manager"  = "argonaut.dev"
          + "argonaut.dev/name"     = "ecredisvalidate10"
          + "argonaut.dev/type"     = "Elasticache Redis"
        }
      + vpc_id                 = "vpc-05233dd009885aadf"
      + timeouts {
          + create = "10m"
          + delete = "15m"
        }
    }
  # module.redis.module.aws_security_group.aws_security_group_rule.keyed["extra[0]"] will be created
  + resource "aws_security_group_rule" "keyed" {
      + cidr_blocks              = [
          + "0.0.0.0/0",
        ]
      + description              = "Allow all outbound traffic"
      + from_port                = 0
      + id                       = (known after apply)
      + prefix_list_ids          = []
      + protocol                 = "-1"
      + security_group_id        = (known after apply)
      + security_group_rule_id   = (known after apply)
      + self                     = false
      + source_security_group_id = (known after apply)
      + to_port                  = 65535
      + type                     = "egress"
    }
  # module.redis.module.aws_security_group.aws_security_group_rule.keyed["extra[1]"] will be created
  + resource "aws_security_group_rule" "keyed" {
      + description              = "Allow all inbound traffic from trusted Security Groups"
      + from_port                = 0
      + id                       = (known after apply)
      + prefix_list_ids          = []
      + protocol                 = "-1"
      + security_group_id        = (known after apply)
      + security_group_rule_id   = (known after apply)
      + self                     = false
      + source_security_group_id = "sg-049551cb7f2616023"
      + to_port                  = 65535
      + type                     = "ingress"
    }
  # module.redis.module.aws_security_group.aws_security_group_rule.keyed["extra[2]"] will be created
  + resource "aws_security_group_rule" "keyed" {
      + cidr_blocks              = [
          + "10.0.0.0/8",
          + "172.0.0.0/8",
          + "192.0.0.0/8",
        ]
      + description              = "Allow connections from within the VPC"
      + from_port                = 0
      + id                       = (known after apply)
      + prefix_list_ids          = []
      + protocol                 = "-1"
      + security_group_id        = (known after apply)
      + security_group_rule_id   = (known after apply)
      + self                     = false
      + source_security_group_id = (known after apply)
      + to_port                  = 6379
      + type                     = "ingress"
    }
  # module.redis.module.security_group.aws_security_group.default[0] will be destroyed
  # (because aws_security_group.default is not in configuration)
  - resource "aws_security_group" "default" {
      - arn                    = "arn:aws:ec2:us-east-2:170982004044:security-group/sg-0c34d941a010d7315" -> null
      - description            = "ElastiCache Security Group" -> null
      - egress                 = [
          - {
              - cidr_blocks      = [
                  - "0.0.0.0/0",
                ]
              - description      = "Allow all outbound traffic"
              - from_port        = 0
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "-1"
              - security_groups  = []
              - self             = false
              - to_port          = 0
            },
        ] -> null
      - id                     = "sg-0c34d941a010d7315" -> null
      - ingress                = [
          - {
              - cidr_blocks      = [
                  - "10.0.0.0/8",
                  - "172.0.0.0/8",
                  - "192.0.0.0/8",
                ]
              - description      = "Allow connections from within the VPC"
              - from_port        = 0
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "-1"
              - security_groups  = []
              - self             = false
              - to_port          = 0
            },
          - {
              - cidr_blocks      = []
              - description      = "Allow all inbound traffic from trusted Security Groups"
              - from_port        = 0
              - ipv6_cidr_blocks = []
              - prefix_list_ids  = []
              - protocol         = "-1"
              - security_groups  = [
                  - "sg-049551cb7f2616023",
                ]
              - self             = false
              - to_port          = 0
            },
        ] -> null
      - name                   = "ecredisvalidate10" -> null
      - owner_id               = "170982004044" -> null
      - revoke_rules_on_delete = false -> null
      - tags                   = {
          - "Name" = "ecredisvalidate10"
        } -> null
      - tags_all               = {
          - "Name"                  = "ecredisvalidate10"
          - "argonaut.dev/env/dev3" = "true"
          - "argonaut.dev/manager"  = "argonaut.dev"
          - "argonaut.dev/name"     = "ecredisvalidate10"
          - "argonaut.dev/type"     = "Elasticache Redis"
        } -> null
      - vpc_id                 = "vpc-05233dd009885aadf" -> null
    }
  # module.redis.module.security_group.aws_security_group_rule.default["egress--1-0-65535-9c87e5e1ed040a443ce1ac8e6d6cf159"] will be destroyed
  # (because aws_security_group_rule.default is not in configuration)
  - resource "aws_security_group_rule" "default" {
      - cidr_blocks            = [
          - "0.0.0.0/0",
        ] -> null
      - description            = "Allow all outbound traffic" -> null
      - from_port              = 0 -> null
      - id                     = "sgrule-3093120347" -> null
      - protocol               = "-1" -> null
      - security_group_id      = "sg-0c34d941a010d7315" -> null
      - security_group_rule_id = "sgr-00143ff366630821d" -> null
      - self                   = false -> null
      - to_port                = 0 -> null
      - type                   = "egress" -> null
    }
  # module.redis.module.security_group.aws_security_group_rule.default["ingress--1-0-6379-8b1feea07de32eb464251c8e610cb6cb"] will be destroyed
  # (because aws_security_group_rule.default is not in configuration)
  - resource "aws_security_group_rule" "default" {
      - cidr_blocks       = [
          - "10.0.0.0/8",
          - "172.0.0.0/8",
          - "192.0.0.0/8",
        ] -> null
      - description       = "Allow connections from within the VPC" -> null
      - from_port         = 0 -> null
      - id                = "sgrule-4161717994" -> null
      - protocol          = "-1" -> null
      - security_group_id = "sg-0c34d941a010d7315" -> null
      - self              = false -> null
      - to_port           = 0 -> null
      - type              = "ingress" -> null
    }
  # module.redis.module.security_group.aws_security_group_rule.default["ingress--1-0-65535-2f9a2d36245ef270e7cdc03d16dd9bf8"] will be destroyed
  # (because aws_security_group_rule.default is not in configuration)
  - resource "aws_security_group_rule" "default" {
      - description              = "Allow all inbound traffic from trusted Security Groups" -> null
      - from_port                = 0 -> null
      - id                       = "sgrule-2452416578" -> null
      - protocol                 = "-1" -> null
      - security_group_id        = "sg-0c34d941a010d7315" -> null
      - security_group_rule_id   = "sgr-0e15b1f45ad659b88" -> null
      - self                     = false -> null
      - source_security_group_id = "sg-049551cb7f2616023" -> null
      - to_port                  = 0 -> null
      - type                     = "ingress" -> null
    }
Plan: 4 to add, 5 to change, 4 to destroy.
```

**Comment by suryaoruganti** [comment]
Plan looks fine. Thanks

**Comment by shubhamgrg04** [comment]
> Plan looks fine. Thanks

Let's merge it then, pls approve the pr

**Review by suryaoruganti — APPROVED** [review]
